hoot-core/src/main/cpp/hoot/core/io/OsmApiChangeset.cpp

```diff
@@ -1496,11 +1539,13 @@ bool XmlChangeset::fixElement(ChangesetTypeMap& map, long id, long version, QMap
           tags.remove(key);
       }
       //  Add in any tags that are missing
-      QXmlStreamAttributes attributes;
       for (QMap<QString, QString>::iterator it = tags.begin(); it != tags.end(); ++it)
-        attributes.append("", it.key(), it.value());
-      if (attributes.size() > 0)
-        element->addTag(XmlObject("tag", QXmlStreamAttributes()));
+      {
+        QXmlStreamAttributes attributes;
+        attributes.append("", "k", it.key());
+        attributes.append("", "v", it.value());
+        element->addTag(XmlObject("tag", attributes));
+      }
```

Comment:
The element that has its version fixed will also have any missing tags from recent updates merged into it, preserving as much of the prior changes as possible.